src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -223,14 +231,60 @@ internal MetricPointStatus MetricPointStatus
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public readonly HistogramBuckets GetHistogramBuckets()
         {
-            if (this.aggType != AggregationType.Histogram && this.aggType != AggregationType.HistogramSumCount)
+            if (this.aggType != AggregationType.Histogram &&
+                this.aggType != AggregationType.HistogramSumCount &&
+                this.aggType != AggregationType.HistogramMinMax &&
+                this.aggType != AggregationType.HistogramSumCountMinMax)
             {
                 this.ThrowNotSupportedMetricTypeException(nameof(this.GetHistogramBuckets));
             }
 
             return this.histogramBuckets;
         }
 
+        /// <summary>
+        /// Gets the minimum value of the histogram associated with the metric
+        /// point if present. Check by using <see cref="HasMinMax"/>.
+        /// </summary>
+        /// <remarks>
+        /// Applies to <see cref="MetricType.Histogram"/> metric type.
+        /// </remarks>
+        /// <returns>A histogram's maximum value.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public double GetHistogramMin()
+        {
+            return this.histogramBuckets.SnapshotMin;
+        }
+
+        /// <summary>
+        /// Gets the maximum value of the histogram associated with the metric
+        /// point if present. Check by using <see cref="HasMinMax"/>.
+        /// </summary>
+        /// <remarks>
+        /// Applies to <see cref="MetricType.Histogram"/> metric type.
+        /// </remarks>
+        /// <returns>A histogram's maximum value.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public double GetHistogramMax()
+        {
+            return this.histogramBuckets.SnapshotMax;
+        }
+
+        /// <summary>
+        /// Gets whether the histogram has a valid Min and Max value.
+        /// Should be used before
+        /// <see cref="GetHistogramMin"/> and <see cref="GetHistogramMax"/> to
+        /// ensure a valid value is returned.
+        /// </summary>
+        /// <returns>A minimum and maximum value exist.</returns>
+        public bool HasMinMax()
+        {
+            return (this.aggType == AggregationType.HistogramMinMax ||
+                   this.aggType == AggregationType.HistogramSumCountMinMax) &&
+                   !(this.histogramBuckets.RunningMin == double.PositiveInfinity &&
```

Comment:
do we actually need to check for +inf,-inf? I think just aggType check is sufficient.

Reply:
This is making sure we have actually seen a value. Would there be an easier way to check that like histogram count > 0?
Actually, checking this is actually not correct since both running min and max will be reset to +inf and -inf after snapshot is called. Any suggestions?

Reply:
> do we actually need to check for +inf,-inf? I think just aggType check is sufficient.

Ah interesting, yes I think you're right. I was originally thinking +inf/-inf would be the meaning of `HasMinMax`, but now that the logic is pushed down then just `AggregationType` can be used.

> Actually, checking this is actually not correct since both running min and max will be reset to +inf and -inf after snapshot is called. Any suggestions?

I think this is ok. From the standpoint of an exporter, all the histogram metric points it received on export that have min/max _should_ be set by that point. I can't think of a scenario where it would still remain unset.

Reply:
^ is correct in my understanding too. @mic-max please remove the +inf and -inf check, and i think we can merge after that.